tests/macro/html-component-fail.stderr

```diff
@@ -36,7 +36,7 @@ error: unexpected token
 58 |     html! { <Child with props ref=() ref=() /> };
    |                                      ^^^
 
-error: unexpected token
+error: Using special syntax `with props` along with named prop is not allowed
```

Comment:
I just realized that this error is misleading... It's not clear from this error message that the following is valid:

```rust
html! { <Child with props ref=() /> } 
```

This is because `ref=..` is a special prop and can be used together with `with props` as long as it comes after. Note, I'm open to `html! { <Child ref=() with props /> } ` being valid as well.

Reply:
Sorry, I got confused which test case that error was for.

In any case, I think that for
```rust
html! { <Child with props ref=() ref=() /> };
```
we should use this error: `"too many refs set"`

Reply:
@jstarry Could you please check if logic is correct?
It is looks like I've covered all possible cases.

I know, the code is a little bit messy, I will fix it.